### PR TITLE
UHF-9576 linked events keywords

### PR DESCRIPTION
**Pull request by hyrsky**
# [UHF-9576](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9576)

## What was done

* Add keyword filter to event_list paragraph.

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-9530-linked-events-keywords`
  * `composer require drupal/helfi_platform_config:dev-UHF-9576-linked-events-keywords`
* Run `make drush-updb drush-cr`
* Update translations: `drush locale:check && drush locale:update; drush cr`.

## How to test

* [x] Run `drush mim linked_events_keywords --reset-threshold 1`
* [x] Check that event list paragraph works with the new features

## Designers review

* [ ] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

## Other PRs

* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/710
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/712

[UHF-9530]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9530?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UHF-9576]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9576?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hyrsky** [comment]
This should be: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9576.

**Review by teroelonen — DISMISSED** [review]
This works nicely. Needs translations but otherwise perfert!

File: src/js/react/apps/linkedevents/components/TopicsFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Select } from 'hds-react';
```
Comment (juho-lehmonen) [inline src/js/react/apps/linkedevents/components/TopicsFilter.tsx:1]: The selected topics aren't added to the selections list. The location filter implementation in `containers/SelectionsContainer.tsx` should match this pretty closely.

File: src/js/react/apps/linkedevents/components/TopicsFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const topics = useAtomValue(topicsAtom);
+  const [topicSelection, setTopicsFilter] = useAtom(topicSelectionAtom);
+  const updateParams = useSetAtom(updateParamsAtom);
+
+  const onChange = (value: any) => {
+    setTopicsFilter(value);
+    updateParams({ [ApiKeys.KEYWORDS]: value.map((topic: any) => topic.value).join(',') });
+  };
+
+  const locationHelper = Drupal.t('Filter events by topics', {}, { context: 'React search topics filter' });
```
Comment (juho-lehmonen) [inline src/js/react/apps/linkedevents/components/TopicsFilter.tsx:18]: For some reason the string context doesn't get picked up by Drupal when the Drupal.t function is called here, so they're visible in the UI translation but without the context. Which means the translations in the `.po` files don't get imported automatically.

Seems to happen in other react components when the translated string is stored in a variable, but works correctly when called directly, like inside the Select component below: `placeholder={Drupal.t('All', {}, { context: 'React search: all available options' })}`

**Review by juho-lehmonen — CHANGES_REQUESTED** [review]

File: src/js/react/apps/linkedevents/components/TopicsFilter.tsx (hunk 1 above)
Comment (hyrsky) [reply src/js/react/apps/linkedevents/components/TopicsFilter.tsx:1]: Good catch, thanks! I was not familiar with this react code before so I missed `SelectionsContainer`.

File: src/js/react/apps/linkedevents/components/TopicsFilter.tsx (hunk 2 above)
Comment (hyrsky) [reply src/js/react/apps/linkedevents/components/TopicsFilter.tsx]: Translations worked for me after running: `drush locale:check && drush locale:update; drush cr`.

**Review by juho-lehmonen — APPROVED** [review]

**Comment by hyrsky** [comment]
> This works nicely. Needs translations but otherwise perfert!

Translations were added here: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/712

**Pull request merged**